modules/lightstep/src/main/scala/Lightstep.scala

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2019 by Rob Norris
+// This software is licensed under the MIT License (MIT).
+// For more information see LICENSE or https://opensource.org/licenses/MIT
+
+package natchez
+package lightstep
+
+import cats.effect.{ Resource, Sync }
+import cats.syntax.applicative._
+import com.lightstep.tracer.jre.JRETracer
+import com.lightstep.tracer.shared.Options
+import io.opentracing.Tracer
+import io.opentracing.propagation.{ Format, TextMapAdapter }
+
+import scala.collection.JavaConverters._
+
+object Lightstep {
+  final case class Configuration(
+    accessToken: String,
+    componentName: String,
+    collectorHost: String,
+    collectorProtocol: String,
+    collectorPort: Int
+  )
+
+  def entryPoint[F[_]: Sync](config: Configuration): Resource[F, EntryPoint[F]] =
```

Comment:
I think maybe we should take an `OptionsBuilder => F[Tracer]` here, like we do with Jaeger, so we're not in the business of configuring LightStep ourselves. Does that makes sense?

Reply:
Actually, it does, as it is even more flexible. However, I would accept only the `OptionsBuilder` and change the Jaeger to accept the `Configuration` as service name can still be overridden which kinda defeats its purpose.

Reply:
Once I think about it, I'll go with `OptionsBuilder => F[Tracer]`, but without setting any parameters during creation.